pkg/build/api/v1/types.go

```diff
@@ -176,8 +176,94 @@ type BuildStatus struct {
 
 	// output describes the Docker image the build has produced.
 	Output BuildStatusOutput `json:"output,omitempty" protobuf:"bytes,10,opt,name=output"`
+
+	// stages contains details about each stage that occurs during the build
+	// including start time, duration (in milliseconds), and the steps that
+	// occured within each stage.
+	Stages []StageInfo `json:"stages,omitempty" protobuf:"bytes,11,opt,name=stages"`
+}
+
+// StageInfo contains details about a build stage.
+type StageInfo struct {
+	// name is a unique identifier for each build stage that occurs.
+	Name StageName `json:"name,omitempty" protobuf:"bytes,1,opt,name=name"`
+
+	// startTime is a timestamp representing the server time when this Stage started.
+	// It is represented in RFC3339 form and is in UTC.
+	StartTime unversioned.Time `json:"startTime,omitempty" protobuf:"bytes,2,opt,name=startTime"`
+
+	// durationMilliseconds identifies how long the stage took
+	// to complete in milliseconds.
+	DurationMilliseconds int64 `json:"durationMilliseconds,omitempty" protobuf:"varint,3,opt,name=durationMilliseconds"`
+
+	// steps contains details about each step that occurs during a build stage
+	// including start time and duration in milliseconds.
+	Steps []StepInfo `json:"steps,omitempty" protobuf:"bytes,4,opt,name=steps"`
 }
 
+// StageName is the unique identifier for each build stage.
+type StageName string
+
+// Valid values for StageName
+const (
```

Comment:
we should remove the ones for BuildPhase too.  Nothing in the code references them so there's no guarantee we use those values for anything that a user would see.

BuildSourceType makes some sense since it represents a valid field value for a client that is constructing a build object so i'm less sure about that one, but i'd be inclined to remove it too.

@smarterclayton @liggitt what's our policy around stuff like this (const values)... are we supposed to define them on the external types.go so they look like part of the api(even though those aren't actually the constants we use in code) or can we just define it on the internal one (which is what we use in code)?